File: openbis_api/source/java/ch/ethz/sis/openbis/generic/shared/api/v3/dto/id/space/SpacePermId.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.shared.api.v3.dto.id.space;

import ch.ethz.sis.openbis.generic.shared.api.v3.dto.id.ObjectPermId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * Space perm id.
 * 
 * @author pkupczyk
 */
@JsonObject("SpacePermId")
public class SpacePermId extends ObjectPermId implements ISpaceId
{

    private static final long serialVersionUID = 1L;

    /**
     * @param permId Space perm id, e.g. "/MY_SPACE" or "MY_SPACE".
     */
    public SpacePermId(String permId)
    {
        super(permId);
    }

    //
    // JSON-RPC
    //

    @SuppressWarnings("unused")
    private SpacePermId()
    {
        super();
    }

    @Override
    public String getPermId()
    {
        String permId = super.getPermId();

        // support both "/MY_SPACE" and "MY_SPACE"
        if (permId.startsWith("/"))
        {
            return permId.substring(1);
        } else
        {
            return permId;
        }
    }

}

File: js-test/servers/common/core-plugins/tests/1/as/webapps/openbis-v3-api-test/html/models/dto/id/space/SpacePermId.js
/**
 *  Space perm id.
 *  
 *  @author pkupczyk
 */
define(["support/stjs", "dto/id/ObjectPermId", "dto/id/space/ISpaceId"], function (stjs, ObjectPermId, ISpaceId) {
    var SpacePermId = /**
     *  @param permId Space perm id, e.g. "/MY_SPACE" or "MY_SPACE".
     */
    function(permId) {
        ObjectPermId.call(this, permId);
    };
    stjs.extend(SpacePermId, ObjectPermId, [ObjectPermId, ISpaceId], function(constructor, prototype) {
        prototype['@type'] = 'dto.id.space.SpacePermId';
        constructor.serialVersionUID = 1;
        prototype.getPermId = function() {
            var permId = ObjectPermId.prototype.getPermId.call(this);
            if (permId.startsWith("/")) {
                return permId.substring(1);
            } else {
                return permId;
            }
        };
    }, {});
    return SpacePermId;
})